src/libstore/store-api.hh

```diff
@@ -158,12 +158,18 @@ public:
 
     const Setting<int> pathInfoCacheSize{this, 65536, "path-info-cache-size", "size of the in-memory store path information cache"};
 
-    const Setting<bool> isTrusted{this, false, "trusted", "whether paths from this store can be used as substitutes even when they lack trusted signatures"};
+    const Setting<bool> isTrusted{this, false, "trusted", "whether paths from this store can be used as substitutes even when they lack trusted signatures. Compare \"trusting\""};
+
+    Setting<bool> isTrusting{this, true, "trusting", "whether (we think) paths can be added to this store even when they lack trusted signatures. Compare \"trusted\""};
```

Comment:
Could we just ask the builder whether they trust us, eliminating the need for a setting?

Reply:
We could but this a good bit more work. Think of the complicated case where one sshes and then the `nix-daemon --stdio` talks to another deaemon. Need a new `Store` interface that all the stores implement. Also, we should really keep it signature based---never mind who "you" are, what is the public key you have signatures for (whether you downloaded or made yourself)? That is the "right way" to do it, and I wouldn't want to do anything but the right away in the protocols where evolution is costly, but that's more work in addition.

My own reservation is we don't have anything like "experimental settings" yet, which would be ideal so we can switch to the stuff above at a later point.

Reply:
Maybe I'm misunderstanding this then. When you set this to true & you're not a trusted-user on that remote store, you get an error right? I just want to ask the daemon whether it will give us an error. It's a new op, but doesn't need an experimental flag or anything. It's the daemon that decides which mode you can or cannot use.

Reply:
You actually don't need a new op necessarily, but could wrap it in try catch statements.

Reply:
You're always going to prefer "true" if you can do it right? So maybe it should be opt-in, but just like `allow-trustless-remote-builds`.

Reply:
Using https://github.com/NixOS/nix/pull/7515 we have gotten rid of it!